File: src/gestorAplicacion/Reserva.java
package gestorAplicacion;

import java.util.LinkedList;

public class Reserva {
    private int numreserva;
    private String fechareserva;
    private Cliente cliente;
    private boolean aplicable;
	public static LinkedList<Reserva> reservas= new LinkedList<>();
     //si la reserva aun es vigente o por el contrario ya vencio

	public Reserva(int numreserva, String fechareserva, Cliente cliente, boolean aplicable) {
	    this.numreserva = numreserva;
	    this.fechareserva = fechareserva;
     	this.aplicable = aplicable;
    	this.cliente = cliente;
	}


    public Cliente getCliente() {
	return cliente;
}


    public void setCliente(Cliente cliente) {
	this.cliente = cliente;
}

	public static void crearReserva(int numreserva, String fechareserva,Cliente cliente, boolean aplicable) {
		Reserva newReserva=new Reserva(numreserva,fechareserva,cliente,aplicable);
		reservas.add(newReserva);
	}

    public int getNumreserva() {
	return numreserva;
}

    public void setNumreserva(int numreserva) {
	this.numreserva = numreserva;
}

    public String getFechareserva() {
	return fechareserva;
}

    public void setFechareserva(String fechareserva) {
	this.fechareserva = fechareserva;
}

    public boolean isAplicable() {
	return aplicable;
}

    public void setAplicable(boolean aplicable) {
	this.aplicable = aplicable;
}

	public static void verReserva(){
		for (Reserva reserva:reservas){
			System.out.println(reserva);
		}
	}

	@Override
	public String toString() {
		return "Nro de Reserva " + numreserva + "\n" +
				"[ Fecha de reserva: " + fechareserva + "\n" +
				"  Cliente: " + cliente + "\n" +
				"  Aplicable: " + aplicable +" ]";
	}

}

File: Python/src/gestorAplicacion/reserva.py
class Reserva:

    reservas={}
    def __init__(self, cliente, numreserva,fechareserva, horareserva,cantidadpersonas):
        self.cliente = cliente
        self.numreserva=numreserva
        self.fechareserva=fechareserva
        self.horareserva=horareserva
        self.cantidadpersonas=cantidadpersonas
        Reserva.reservas[numreserva]=self
    

    #getters y setters   
    def setCliente(self, cliente):
        self.cliente = cliente
    
    def getCliente(self):
        return self.cliente

    def setNumreserva(self, numreserva):
        self.numreserva=numreserva
    
    def getNumreserva(self):
        return self.numreserva

    def setFechareserva(self, fechareserva):
        self.fechareserva=fechareserva
    
    def getFechareserva(self):
        return self.fechareserva

    def setHorareserva(self, horareserva):
        self.horareserva=horareserva
    
    def getHorareserva(self):
        return self.horareserva

    def setCantidadpersonas(self, cantidadpersonas):
        self.cantidadpersonas=cantidadpersonas
    
    def getCantidadpersonas(self):
        return self.cantidadpersonas


    